src/delagent/agent/util.c

```diff
@@ -986,3 +996,104 @@ void Usage (char *Name)
   fprintf(stderr,"  --user|-n # :: user name\n");
   fprintf(stderr,"  --password|-p # :: password\n");
 } /* Usage() */
+
+/**
+ * \brief check if the uploads in a folder can be deleted, that is the user have
+ * the permissin to delte this the child folders aswell
+ * 
+ * \param long parent - folder_pk
+ * \param char user_id - user_id
+ * 
+ * \return 0: cannot be deleted 1: can be deleted;
+ */
+int check_upload_folder_del(long parent, int user_id)
+{
+  char SQL[MAXSQL];
+  PGresult *result;
+  int count = 0;
+  int i = 0;
+  long upload_id = 0;
+
+  memset(SQL,'\0',sizeof(SQL));
+  snprintf(SQL,sizeof(SQL),"SELECT upload_pk FROM folderlist WHERE folder_pk is null AND parent = %ld;",parent);
+  result = PQexec(db_conn, SQL);
+  if (fo_checkPQresult(db_conn, result, SQL, __FILE__, __LINE__)) exit(-1);
+  count = PQntuples(result);
+  if (count == 0){
+    return 1;
+  }
+  for(i = 0; i < count; i++){
+    upload_id = atol(PQgetvalue(result, 0, i));
```

Comment:
The function `PQgetvalue` is of the form
``` 
char *PQgetvalue(const PGresult *res, int row_number, int column_number);
```
this means that you are iterating through the columns. Is that what you want? (`PQntuples` returns the number of rows) (see [link](http://www.postgresql.org/docs/8.4/static/libpq-exec.html))

The same applies to [line 1060](https://github.com/fossology/fossology/pull/602/files#diff-020758a99280d21bcd094cf5f770500bR1060).